### ISSUE TITLE
Version mapbox sdk

### ISSUE THREAD

**Issue by wuilmerj24** [EddyVerbruggen/nativescript-mapbox#330]
I am creating a plugin for Mapbox Navigations ui. But when implementing, they give incompatibility errors. And resolving that incompatibility is a lot of work. What version of mapbox will you use in the new version of this plugin?

**Comment by Yermo** [comment]
My plan is to use the current version.

**Comment by wuilmerj24** [comment]
@Yermo Okay. Is it already in development?

**Comment by Yermo** [comment]
Yes. I have it working but have a bit more work to do before it’s ready to merge.

My fork introduces some API changes but they are not major.

**Comment by wuilmerj24** [comment]
@Yermo Do you have a beta or alpha version?

**Comment by Yermo** [comment]
I just committed my latest changes. It's still a work in progress but you can check out my progress here: https://github.com/Yermo/nativescript-mapbox

The documentation is not yet updated so refer to the comments in demos and the source.

There is still a random crash on Android when navigating away from a map in an Angular app. I have spent crazy hours trying to track it down but have not yet identified the cause of the crash. There is an e2e test in the demo-angular app that exercises that flaw. Any help tracking that down would be greatly appreciated.

How maps are displayed programmatically has been changed and a couple of API's have changed. See the demo.

I have to move over a few changes from the main repository, notably addSource() and make a few other updates, and update the documentation, then it should be close to read to merge.

**Comment by wuilmerj24** [comment]
@Yermo Hi. What percentage is missing to finish?

**Comment by Yermo** [comment]
All that needs to be done is to update the Android geolocation code so that all the features of the demo work. I figure it would take me maybe a day or two, if that, to come up to speed on what changed and when and how to fold in whatever the new API is. 

Less less than 1%. 1 feature and then @EddyVerbruggen says he will likely accept the pull request. 

I think I'm about 2 or 3 weeks out from being able to focus it (fingers crossed).

**Comment by wuilmerj24** [comment]
@Yermo okay.

**Comment by wuilmerj24** [comment]
@Yermo Hi. Have you been able to merge?

**Comment by Yermo** [comment]
@wuilmerj24 I've put some time into it but I have not finished it yet. In the mean time, you can use my fork. Any code written against my fork should work against the upcoming merge.

**Comment by wuilmerj24** [comment]
@Yermo Hi. Already the problem that it presents with respect to android could you correct it?

**Comment by Yermo** [comment]
@wuilmerj24 I do not understand your question. I am working on the merge.

**Comment by wuilmerj24** [comment]
@Yermo A few months ago the map crashed on android. The crash was generated when you were navigating back to the screen where you had the map.

**Comment by Yermo** [comment]
@wuilmerj24 This is not a bug in the Mapbox plugin but instead a bug in tns-core-modules https://github.com/NativeScript/NativeScript/issues/7867#issuecomment-569230402

**Comment by wuilmerj24** [comment]
@Yermo Okay. Thanks.
